### PR TITLE
fix(FEC-7594): native adapter keeps buffering when connection is lost

### PR DESCRIPTION
**Pull request by odedhutzler**
When there is no network and native adapter is used, the video element keeps buffering forever (the video element state stays '`waiting`'). 

the problem is that no error is thrown, so the player doesn't know there is an issue and showing the spinner forever.

the suggested fix starts a timeout upon a '`waiting`' event and checking if the browser is offline.
the timeout is cleared if we are playing / pausing etc.

### CheckLists

- [x] changes have been done against master branch, and PR does not conflict
- [ ] new unit / functional tests have been added (whenever applicable)
- [x] test are passing in local environment 
- [ ] Travis tests are passing (or test results are not worse than on master branch :))
- [ ] Docs have been updated

### CODE REVIEW

File: src/engines/html5/media-source/adapters/native-adapter.js [hunk 1]
```diff
@@ -283,6 +289,33 @@ export default class NativeAdapter extends BaseMediaSourceAdapter {
...
+          )
+        );
+      } else {
+        this._bufferTimeoutId = setTimeout(onTimeOut, WAIT_TIME);
+      }
+    };
+    this._bufferTimeoutId = setTimeout(onTimeOut, WAIT_TIME);
+  }
+
+  _clearBufferTimeout(): void {
```
Comment (yairans) [inline src/engines/html5/media-source/adapters/native-adapter.js]: consider `_clearWaitingTimeout`

File: src/engines/html5/media-source/adapters/native-adapter.js [hunk 2]
```diff
@@ -283,6 +289,33 @@ export default class NativeAdapter extends BaseMediaSourceAdapter {
...
+    if (this._bufferTimeoutId) {
+      clearTimeout(this._bufferTimeoutId);
```
Comment (yairans) [inline src/engines/html5/media-source/adapters/native-adapter.js]: `this._bufferTimeoutId = null`

File: src/engines/html5/media-source/adapters/native-adapter.js [hunk 3]
```diff
@@ -283,6 +289,33 @@ export default class NativeAdapter extends BaseMediaSourceAdapter {
...
+          Html5EventType.ERROR,
+          new Error(
+            Error.Severity.CRITICAL,
+            Error.Category.NETWORK,
+            Error.Code.TIMEOUT,
+            `player exceeded ${WAIT_TIME} timeout and connection seems to be offline`
+          )
+        );
+      } else {
+        this._bufferTimeoutId = setTimeout(onTimeOut, WAIT_TIME);
```
Comment (yairans) [inline src/engines/html5/media-source/adapters/native-adapter.js]: use `setInterval` instead of nested `setTimeout`

**Comment by OrenMe** [comment]
@yairans please update review pending @odedhutzler's changes.

**Pull request merged**